src/erl.rs

```diff
@@ -321,23 +321,30 @@ fn const_segment(
         _ => None,
     };
 
-    bit_string_segment(document, options, size, unit, env)
+    bit_string_segment(document, options, size, unit, false, env)
 }
 
 fn expr_segment(
     value: &TypedExpr,
     options: Vec<BitStringSegmentOption<TypedExpr>>,
     env: &mut Env,
 ) -> Document {
+    let mut value_is_a_variable = false;
+
     let document = match value {
         // Skip the normal <<value/utf8>> surrounds
         TypedExpr::String { value, .. } => value.clone().to_doc().surround("\"", "\""),
 
         // As normal
-        TypedExpr::Int { .. }
-        | TypedExpr::Float { .. }
-        | TypedExpr::Var { .. }
-        | TypedExpr::BitString { .. } => expr(value, env),
+        TypedExpr::Int { .. } | TypedExpr::Float { .. } | TypedExpr::BitString { .. } => {
+            expr(value, env)
+        }
+
+        // Return value as normal, but set flag
+        TypedExpr::Var { .. } => {
+            value_is_a_variable = true;
```

Comment:
I think we may be interested in whether it is a literal rather than when it a variable as I think the same problem may occur for function calls, tuple indexing and field accessing.

Reply:
Oh yeah, of course. Will do.

Reply:
Thank you